Flaky test: `Im2Col`

https://jenkins.preferred.jp/job/chainer/job/chainer_pr/1962/TEST=chainer-py3,label=mn1-p100/console

```python
15:46:32 _ TestIm2Col_param_21_{ksize=(2, 3), stride=(1, 2), pad=0, dilate=(2, 1), cover_all=True, dtype=float16}.test_backward_gpu _
15:46:32 
15:46:32 self = <chainer.testing._bundle.TestIm2Col_param_21_{ksize=(2, 3), stride=(1, 2), pad=0, dilate=(2, 1), cover_all=True, dtype=float16} testMethod=test_backward_gpu>
15:46:32 
15:46:32     @attr.gpu
15:46:32     def test_backward_gpu(self):
15:46:32         self.check_backward(
15:46:32             cuda.to_gpu(self.x), self.ksize, self.stride, self.pad,
15:46:32 >           self.cover_all, self.dilate, cuda.to_gpu(self.gy))
15:46:32 
15:46:32 self       = <chainer.testing._bundle.TestIm2Col_param_21_{ksize=(2, 3), stride=(1, 2), pad=0, dilate=(2, 1), cover_all=True, dtype=float16} testMethod=test_backward_gpu>
15:46:32 
15:46:32 tests/chainer_tests/functions_tests/array_tests/test_im2col.py:146: 
15:46:32 _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
15:46:32 tests/chainer_tests/functions_tests/array_tests/test_im2col.py:135: in check_backward
15:46:32     f, x, gy, dtype=numpy.float64, **self.check_backward_options)
15:46:32 chainer/gradient_check.py:900: in check_backward
15:46:32     detect_nondifferentiable, is_immutable_params=False
15:46:32 chainer/gradient_check.py:464: in run
15:46:32     self._run()
15:46:32 chainer/gradient_check.py:507: in _run
15:46:32     self._compare_gradients(gx_numeric, gx_backward, directions)
15:46:32 _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
15:46:32 
15:46:32 self = <chainer.gradient_check._CheckBackward object at 0x7f88929616a0>
15:46:32 gx_numeric = array(0.00582334), gx_backward = array(0.00695735)
15:46:32 directions = [array([[[[-1.32758054e-04,  2.82382409e-02,  1.08486095e-03,
15:46:32            7.54827629e-02, -3.47154825e-02, -8.02623359e...  [ 2.01067435e-03, -3.04967542e-02,  8.04134911e-02,
15:46:32            1.55760730e-02,  2.93159456e-02, -7.29616962e-03]]]])]
15:46:32 
15:46:32     def _compare_gradients(self, gx_numeric, gx_backward, directions):
15:46:32         atol = self.atol
15:46:32         rtol = self.rtol
15:46:32         # Compare the gradients
15:46:32         try:
15:46:32             testing.assert_allclose(
15:46:32                 gx_numeric, gx_backward, atol=atol, rtol=rtol)
15:46:32         except AssertionError as e:
15:46:32             eps = self.eps
15:46:32             xs = self.xs
15:46:32             gys = self.gys
15:46:32             f = six.StringIO()
15:46:32             f.write('check_backward failed (eps={} atol={} rtol={})\n'.format(
15:46:32                 eps, atol, rtol))
15:46:32             for i, x in enumerate(xs):
15:46:32                 f.write('inputs[{}]:\n'.format(i))
15:46:32                 f.write('{}\n'.format(x))
15:46:32             for i, gy in enumerate(gys):
15:46:32                 f.write('grad_outputs[{}]:\n'.format(i))
15:46:32                 f.write('{}\n'.format(gy))
15:46:32             for i, d in enumerate(directions):
15:46:32                 f.write('directions[{}]:\n'.format(i))
15:46:32                 f.write('{}\n'.format(d))
15:46:32             f.write('gradients (numeric):  {}\n'.format(gx_numeric))
15:46:32             f.write('gradients (backward): {}\n'.format(gx_backward))
15:46:32             f.write('\n')
15:46:32             f.write('x: numeric gradient, y: backward gradient')
15:46:32             f.write(str(e))
15:46:32 >           raise AssertionError(f.getvalue())
15:46:32 E           AssertionError: Parameterized test failed.
15:46:32 E           
15:46:32 E           Base test method: TestIm2Col.test_backward_gpu
15:46:32 E           Test parameters:
15:46:32 E             ksize: (2, 3)
15:46:32 E             stride: (1, 2)
15:46:32 E             pad: 0
15:46:32 E             dilate: (2, 1)
15:46:32 E             cover_all: True
15:46:32 E             dtype: <class 'numpy.float16'>
15:46:32 E           
15:46:32 E           
15:46:32 E           (caused by)
15:46:32 E           AssertionError: check_backward failed (eps=0.001 atol=0.001 rtol=0.01)
15:46:32 E           inputs[0]:
15:46:32 E           [[[[0.4407   0.4185   0.778    0.8325   0.4678   0.47    ]
15:46:32 E              [0.884    0.441    0.4668   0.4902   0.9883   0.5054  ]
15:46:32 E              [0.386    0.6816   0.454    0.6616   0.2274   0.791   ]
15:46:32 E              [0.655    0.8706   0.874    0.2886   0.189    0.582   ]
15:46:32 E              [0.653    0.6465   0.615    0.875    0.03177  0.9683  ]
15:46:32 E              [0.664    0.8877   0.5713   0.4753   0.8135   0.1302  ]
15:46:32 E              [0.9937   0.938    0.937    0.532    0.496    0.583   ]
15:46:32 E              [0.2092   0.2822   0.7207   0.1412   0.02977  0.3499  ]]
15:46:32 E           
15:46:32 E             [[0.997    0.011765 0.7495   0.733    0.4583   0.711   ]
15:46:32 E              [0.887    0.5146   0.5493   0.984    0.4272   0.6064  ]
15:46:32 E              [0.131    0.03894  0.7593   0.3506   0.9155   0.01355 ]
15:46:32 E              [0.637    0.566    0.414    0.2344   0.0959   0.7407  ]
15:46:32 E              [0.908    0.212    0.804    0.2227   0.717    0.969   ]
15:46:32 E              [0.67     0.0746   0.722    0.507    0.8354   0.754   ]
15:46:32 E              [0.05188  0.5747   0.783    0.3625   0.599    0.713   ]
15:46:32 E              [0.5493   0.6455   0.747    0.488    0.6274   0.927   ]]
15:46:32 E           
15:46:32 E             [[0.643    0.3037   0.1962   0.3389   0.8545   0.1437  ]
15:46:32 E              [0.6772   0.3333   0.976    0.4958   0.5454   0.885   ]
15:46:32 E              [0.1906   0.9463   0.5894   0.8      0.8457   0.4746  ]
15:46:32 E              [0.806    0.4429   0.6626   0.2085   0.04776  0.3843  ]
15:46:32 E              [0.3132   0.4385   0.7417   0.2177   0.07245  0.8286  ]
15:46:32 E              [0.168    0.6733   0.649    0.5537   0.374    0.834   ]
15:46:32 E              [0.2651   0.909    0.5977   0.9014   0.737    0.507   ]
15:46:32 E              [0.02863  0.3323   0.8633   0.6914   0.739    0.58    ]]]
15:46:32 E           
15:46:32 E           
15:46:32 E            [[[0.00594  0.6445   0.861    0.151    0.755    0.6235  ]
15:46:32 E              [0.559    0.955    0.815    0.133    0.856    0.803   ]
15:46:32 E              [0.5454   0.895    0.783    0.3901   0.00261  0.148   ]
15:46:32 E              [0.1671   0.10596  0.6836   0.596    0.575    0.765   ]
15:46:32 E              [0.4785   0.7207   0.51     0.2382   0.743    0.767   ]
15:46:32 E              [0.949    0.565    0.6646   0.6055   0.579    0.194   ]
15:46:32 E              [0.3098   0.497    0.3801   0.26     0.4907   0.8843  ]
15:46:32 E              [0.506    0.54     0.2559   0.4756   0.1985   0.823   ]]
15:46:32 E           
15:46:32 E             [[0.0957   0.8076   0.8013   0.9536   0.5273   0.0746  ]
15:46:32 E              [0.7085   0.03394  0.01332  0.536    0.72     0.713   ]
15:46:32 E              [0.499    0.8833   0.742    0.2462   0.6772   0.597   ]
15:46:32 E              [0.8086   0.2384   0.94     0.2905   0.794    0.1616  ]
15:46:32 E              [0.2397   0.5835   0.11566  0.9106   0.2627   0.762   ]
15:46:32 E              [0.7407   0.4236   0.06946  0.462    0.828    0.2673  ]
15:46:32 E              [0.3445   0.3943   0.7344   0.7427   0.8145   0.298   ]
15:46:32 E              [0.61     0.509    0.2568   0.1655   0.9575   0.164   ]]
15:46:32 E           
15:46:32 E             [[0.4595   0.06024  0.983    0.12476  0.667    0.10913 ]
15:46:32 E              [0.7153   0.4106   0.007515 0.38     0.731    0.139   ]
15:46:32 E              [0.5225   0.7554   0.341    0.698    0.3037   0.1592  ]
15:46:32 E              [0.475    0.6646   0.2717   0.8296   0.3374   0.01944 ]
15:46:32 E              [0.51     0.429    0.6426   0.5693   0.4443   0.05643 ]
15:46:32 E              [0.0431   0.3093   0.6104   0.0584   0.877    0.7817  ]
15:46:32 E              [0.442    0.639    0.923    0.7905   0.7715   0.569   ]
15:46:32 E              [0.00766  0.6416   0.973    0.536    0.0473   0.9062  ]]]]
15:46:32 E           grad_outputs[0]:
15:46:32 E           [[[[7.2327e-02 8.7646e-01 9.4092e-01]
15:46:32 E              [8.9307e-01 2.7905e-01 7.8906e-01]
15:46:32 E              [6.0059e-01 1.5759e-01 8.1152e-01]
15:46:32 E              [5.7129e-01 9.6973e-01 1.3843e-01]
15:46:32 E              [9.4775e-01 4.1235e-01 1.2952e-01]
15:46:32 E              [7.7246e-01 6.1084e-01 2.3511e-01]]
15:46:32 E           
15:46:32 E             [[5.3955e-01 8.7500e-01 2.6417e-03]
15:46:32 E              [3.5107e-01 3.2764e-01 6.1328e-01]
15:46:32 E              [1.1926e-01 9.0967e-01 4.3335e-01]
15:46:32 E              [5.9424e-01 3.9624e-01 7.7441e-01]
15:46:32 E              [6.3086e-01 4.3433e-01 6.3037e-01]
15:46:32 E              [3.8135e-01 8.8623e-01 5.2393e-01]]
15:46:32 E           
15:46:32 E             [[6.9336e-01 8.2324e-01 4.1089e-01]
15:46:32 E              [6.3428e-01 8.4717e-01 9.5093e-02]
15:46:32 E              [4.9976e-01 4.8315e-01 5.2734e-01]
15:46:32 E              [3.7061e-01 9.5850e-01 9.2432e-01]
15:46:32 E              [5.4248e-01 7.7539e-01 4.0234e-01]
15:46:32 E              [3.4058e-01 3.6230e-01 8.4912e-01]]
15:46:32 E           
15:46:32 E             [[7.1533e-01 5.0342e-01 6.1768e-01]
15:46:32 E              [2.4551e-02 7.9004e-01 5.6702e-02]
15:46:32 E              [6.6699e-01 6.8665e-02 2.7930e-01]
15:46:32 E              [5.3955e-01 4.4019e-01 1.5112e-01]
15:46:32 E              [5.5225e-01 5.6055e-01 6.5479e-01]
15:46:32 E              [8.4521e-01 7.8760e-01 6.1230e-01]]
15:46:32 E           
15:46:32 E             [[2.5977e-01 3.6353e-01 9.2285e-01]
15:46:32 E              [7.5293e-01 9.8291e-01 9.7510e-01]
15:46:32 E              [5.6348e-01 1.4160e-01 7.1240e-01]
15:46:32 E              [8.0371e-01 8.7158e-02 9.0820e-01]
15:46:32 E              [7.7344e-01 5.8350e-01 3.2544e-01]
15:46:32 E              [7.7490e-01 9.0527e-01 1.8896e-01]]
15:46:32 E           
15:46:32 E             [[7.5732e-01 8.3923e-02 3.3594e-01]
15:46:32 E              [3.9825e-02 3.7134e-01 9.5947e-01]
15:46:32 E              [9.5996e-01 3.2178e-01 2.6294e-01]
15:46:32 E              [4.4165e-01 9.2383e-01 7.1240e-01]
15:46:32 E              [7.2607e-01 3.1079e-01 8.6963e-01]
15:46:32 E              [4.5044e-02 4.6094e-01 1.4673e-01]]
15:46:32 E           
15:46:32 E             [[1.7419e-01 3.3472e-01 9.9365e-01]
15:46:32 E              [3.9990e-01 4.2700e-01 1.9421e-01]
15:46:32 E              [7.3340e-01 8.6523e-01 5.7910e-01]
15:46:32 E              [3.4448e-01 1.5356e-01 2.8198e-01]
15:46:32 E              [7.6221e-01 8.3252e-01 9.3506e-01]
15:46:32 E              [9.0967e-01 8.3154e-01 8.5645e-01]]
15:46:32 E           
15:46:32 E             [[3.5254e-01 6.1475e-01 4.2090e-01]
15:46:32 E              [4.1650e-01 2.1533e-01 7.6807e-01]
15:46:32 E              [4.9390e-01 5.9766e-01 5.6348e-01]
15:46:32 E              [6.8506e-01 3.4009e-01 2.3102e-02]
15:46:32 E              [2.9102e-01 5.6641e-01 2.6138e-02]
15:46:32 E              [4.8340e-01 2.3303e-01 7.2852e-01]]
15:46:32 E           
15:46:32 E             [[2.3425e-01 1.7004e-01 6.4502e-01]
15:46:32 E              [9.5020e-01 1.7346e-01 4.0186e-01]
15:46:32 E              [3.4009e-01 5.3955e-02 4.9121e-01]
15:46:32 E              [9.3213e-01 7.1582e-01 3.4351e-01]
15:46:32 E              [5.1855e-01 4.2944e-01 9.3628e-02]
15:46:32 E              [6.5088e-01 2.6392e-01 7.2803e-01]]
15:46:32 E           
15:46:32 E             [[4.7144e-01 8.8928e-02 7.3975e-01]
15:46:32 E              [4.8828e-02 3.4058e-01 8.5449e-01]
15:46:32 E              [4.3286e-01 6.4258e-01 5.8203e-01]
15:46:32 E              [7.5537e-01 8.9111e-01 8.3923e-02]
15:46:32 E              [9.7705e-01 8.7109e-01 7.6111e-02]
15:46:32 E              [9.2676e-01 8.3838e-01 7.8955e-01]]
15:46:32 E           
15:46:32 E             [[6.8555e-01 2.9980e-01 8.1396e-01]
15:46:32 E              [5.5957e-01 9.2285e-01 5.3027e-01]
15:46:32 E              [5.2393e-01 1.9849e-01 4.1699e-01]
15:46:32 E              [8.2324e-01 4.3286e-01 8.2910e-01]
15:46:32 E              [9.6289e-01 3.4256e-03 8.2129e-01]
15:46:32 E              [8.3252e-01 1.4294e-01 5.5664e-01]]
15:46:32 E           
15:46:32 E             [[4.6997e-02 5.7324e-01 5.1270e-01]
15:46:32 E              [8.8379e-01 5.9570e-01 1.1310e-01]
15:46:32 E              [4.1919e-01 3.6621e-01 6.0449e-01]
15:46:32 E              [5.9912e-01 1.1421e-02 8.0127e-01]
15:46:32 E              [8.2373e-01 9.5654e-01 3.6084e-01]
15:46:32 E              [2.4780e-01 7.1338e-01 8.1348e-01]]
15:46:32 E           
15:46:32 E             [[3.9502e-01 2.9346e-01 2.1851e-01]
15:46:32 E              [3.4058e-01 6.8799e-01 8.5986e-01]
15:46:32 E              [7.5342e-01 2.8809e-01 3.0981e-01]
15:46:32 E              [1.0986e-01 8.5059e-01 5.6152e-01]
15:46:32 E              [3.5767e-01 9.2773e-01 9.2285e-01]
15:46:32 E              [5.0781e-01 8.5815e-02 7.6562e-01]]
15:46:32 E           
15:46:32 E             [[3.1812e-01 3.7744e-01 3.0457e-02]
15:46:32 E              [9.1357e-01 2.3743e-01 1.6006e-02]
15:46:32 E              [4.2432e-01 3.9062e-01 6.8457e-01]
15:46:32 E              [9.2236e-01 2.2766e-02 3.6157e-01]
15:46:32 E              [4.0796e-01 5.1172e-01 1.3538e-01]
15:46:32 E              [7.8711e-01 4.9512e-01 6.0596e-01]]
15:46:32 E           
15:46:32 E             [[8.7695e-01 6.7969e-01 2.4878e-01]
15:46:32 E              [7.6758e-01 1.7609e-02 3.8647e-01]
15:46:32 E              [6.2988e-01 6.4331e-02 1.0083e-01]
15:46:32 E              [8.6328e-01 5.3613e-01 1.9775e-01]
15:46:32 E              [5.4639e-01 3.0444e-01 7.9834e-01]
15:46:32 E              [8.8721e-01 9.2383e-01 4.8462e-01]]
15:46:32 E           
15:46:32 E             [[1.6980e-01 8.0078e-01 9.8193e-01]
15:46:32 E              [9.2627e-01 3.2495e-01 3.6914e-01]
15:46:32 E              [2.1716e-01 7.7881e-02 4.2700e-01]
15:46:32 E              [1.6553e-01 9.1113e-01 3.0664e-01]
15:46:32 E              [4.2090e-01 4.9951e-01 6.9141e-01]
15:46:32 E              [4.1528e-01 9.7168e-01 3.2275e-01]]
15:46:32 E           
15:46:32 E             [[2.0117e-01 1.4319e-01 1.0065e-01]
15:46:32 E              [7.2656e-01 1.4648e-01 5.0293e-01]
15:46:32 E              [7.3779e-01 9.9902e-01 2.8003e-01]
15:46:32 E              [1.3049e-01 1.4319e-01 9.6631e-01]
15:46:32 E              [6.2842e-01 1.5076e-01 6.8262e-01]
15:46:32 E              [2.5928e-01 4.9866e-02 9.9609e-01]]
15:46:32 E           
15:46:32 E             [[1.8579e-01 1.4026e-01 7.7734e-01]
15:46:32 E              [5.2490e-01 5.0000e-01 5.8008e-01]
15:46:32 E              [3.1274e-01 1.5845e-01 9.1064e-01]
15:46:32 E              [1.8872e-01 9.0771e-01 6.2695e-01]
15:46:32 E              [5.0293e-01 6.5186e-01 5.5713e-01]
15:46:32 E              [3.7451e-01 3.2056e-01 9.3750e-01]]]
15:46:32 E           
15:46:32 E           
15:46:32 E            [[[7.7783e-01 6.8018e-01 1.9885e-01]
15:46:32 E              [3.0054e-01 4.6851e-01 4.0063e-01]
15:46:32 E              [2.4994e-02 9.0820e-01 7.6270e-01]
15:46:32 E              [6.4258e-01 2.0532e-01 5.6641e-01]
15:46:32 E              [2.2351e-01 1.7212e-01 3.7939e-01]
15:46:32 E              [9.2627e-01 1.3756e-02 5.0439e-01]]
15:46:32 E           
15:46:32 E             [[7.5830e-01 7.4219e-02 2.5781e-01]
15:46:32 E              [4.2725e-01 3.8867e-01 9.2578e-01]
15:46:32 E              [1.6724e-01 3.4741e-01 2.9129e-02]
15:46:32 E              [8.5449e-01 8.9598e-04 4.3701e-01]
15:46:32 E              [7.4561e-01 3.5376e-01 6.7432e-01]
15:46:32 E              [7.5928e-01 3.6621e-01 1.4502e-01]]
15:46:32 E           
15:46:32 E             [[4.6411e-01 9.5166e-01 8.5156e-01]
15:46:32 E              [1.4514e-01 4.0527e-02 9.5947e-02]
15:46:32 E              [7.9443e-01 3.8794e-01 9.1797e-01]
15:46:32 E              [9.2725e-01 9.2920e-01 8.9160e-01]
15:46:32 E              [2.4121e-01 7.6270e-01 4.7241e-02]
15:46:32 E              [6.2012e-01 8.3154e-01 7.5195e-01]]
15:46:32 E           
15:46:32 E             [[4.3311e-01 8.9648e-01 6.8213e-01]
15:46:32 E              [1.6589e-01 7.1826e-01 2.8955e-01]
15:46:32 E              [8.7280e-02 5.3711e-01 5.4736e-01]
15:46:32 E              [4.7485e-01 6.3672e-01 1.9751e-01]
15:46:32 E              [1.2451e-01 1.3989e-01 1.1475e-01]
15:46:32 E              [8.9355e-01 3.9966e-01 8.7842e-01]]
15:46:32 E           
15:46:32 E             [[9.9902e-01 8.5449e-01 3.3691e-01]
15:46:32 E              [6.3135e-01 1.2493e-03 8.2178e-01]
15:46:32 E              [9.0332e-01 7.9883e-01 4.1748e-01]
15:46:32 E              [9.4531e-01 1.6772e-01 1.0541e-01]
15:46:32 E              [3.9160e-01 2.8223e-01 1.7188e-01]
15:46:32 E              [9.6729e-01 1.3989e-01 8.6475e-01]]
15:46:32 E           
15:46:32 E             [[7.1094e-01 3.1323e-01 2.3914e-01]
15:46:32 E              [4.4824e-01 8.9990e-01 2.9028e-01]
15:46:32 E              [4.0985e-02 2.1228e-01 2.2095e-01]
15:46:32 E              [7.7686e-01 5.1611e-01 5.8301e-01]
15:46:32 E              [8.1006e-01 2.7368e-01 4.2871e-01]
15:46:32 E              [7.9688e-01 8.8330e-01 1.8005e-01]]
15:46:32 E           
15:46:32 E             [[2.9614e-01 1.0773e-01 6.9531e-01]
15:46:32 E              [1.4233e-01 2.0874e-01 8.5938e-01]
15:46:32 E              [4.6069e-01 2.6440e-01 5.0293e-01]
15:46:32 E              [9.8999e-02 6.6064e-01 7.1094e-01]
15:46:32 E              [8.9453e-01 2.9175e-01 7.9932e-01]
15:46:32 E              [4.7217e-01 4.8560e-01 9.7607e-01]]
15:46:32 E           
15:46:32 E             [[6.5723e-01 6.8408e-01 9.7559e-01]
15:46:32 E              [5.6836e-01 8.9893e-01 9.1797e-01]
15:46:32 E              [1.2231e-01 5.5176e-02 3.9185e-01]
15:46:32 E              [5.8838e-01 2.1252e-01 8.3594e-01]
15:46:32 E              [6.4795e-01 3.5327e-01 2.8516e-01]
15:46:32 E              [5.8936e-01 7.9785e-01 3.6938e-01]]
15:46:32 E           
15:46:32 E             [[9.0479e-01 6.8213e-01 5.9033e-01]
15:46:32 E              [6.8604e-01 1.1627e-01 7.2852e-01]
15:46:32 E              [8.2227e-01 1.8347e-01 5.4004e-01]
15:46:32 E              [8.0371e-01 4.1919e-01 5.5859e-01]
15:46:32 E              [9.8730e-01 2.8735e-01 4.9170e-01]
15:46:32 E              [7.3877e-01 5.5273e-01 8.4912e-01]]
15:46:32 E           
15:46:32 E             [[9.6484e-01 1.7261e-01 5.9424e-01]
15:46:32 E              [3.8599e-01 7.1777e-01 2.9395e-01]
15:46:32 E              [8.1348e-01 7.5977e-01 7.1240e-01]
15:46:32 E              [4.9609e-01 9.9121e-01 2.0752e-01]
15:46:32 E              [2.7710e-01 8.1543e-01 7.5049e-01]
15:46:32 E              [3.9978e-02 8.6523e-01 1.0828e-01]]
15:46:32 E           
15:46:32 E             [[2.4988e-01 3.2544e-01 4.9170e-01]
15:46:32 E              [6.7749e-02 8.7939e-01 5.0830e-01]
15:46:32 E              [5.5420e-01 5.9143e-02 6.4697e-01]
15:46:32 E              [7.1167e-02 5.1318e-01 3.8916e-01]
15:46:32 E              [8.4082e-01 8.3203e-01 8.9307e-01]
15:46:32 E              [6.0645e-01 1.0889e-01 9.5166e-01]]
15:46:32 E           
15:46:32 E             [[7.8711e-01 2.3535e-01 5.0635e-01]
15:46:32 E              [6.0645e-01 2.4231e-01 3.4253e-01]
15:46:32 E              [3.9062e-01 8.3740e-01 8.9258e-01]
15:46:32 E              [2.4475e-01 9.3506e-01 8.3740e-02]
15:46:32 E              [6.3721e-01 9.1455e-01 2.2778e-01]
15:46:32 E              [2.7115e-02 5.7422e-01 4.4507e-01]]
15:46:32 E           
15:46:32 E             [[4.8267e-01 1.8091e-01 7.0898e-01]
15:46:32 E              [9.3506e-01 8.1104e-01 4.4214e-01]
15:46:32 E              [1.1414e-01 2.2412e-01 1.1401e-01]
15:46:32 E              [5.4980e-01 7.1826e-01 3.1714e-01]
15:46:32 E              [2.3279e-01 1.2988e-01 1.1401e-01]
15:46:32 E              [9.2627e-01 7.5781e-01 7.2754e-01]]
15:46:32 E           
15:46:32 E             [[5.7471e-01 3.0502e-02 5.4248e-01]
15:46:32 E              [8.4717e-01 4.8828e-01 5.9521e-01]
15:46:32 E              [6.7432e-01 1.1276e-02 9.5752e-01]
15:46:32 E              [9.6094e-01 5.3955e-01 5.5664e-01]
15:46:32 E              [2.7832e-01 9.8730e-01 2.5375e-02]
15:46:32 E              [5.7959e-01 6.7444e-02 6.7383e-01]]
15:46:32 E           
15:46:32 E             [[6.1230e-01 4.9219e-01 6.6602e-01]
15:46:32 E              [3.5986e-01 3.6523e-01 4.5654e-01]
15:46:32 E              [3.1592e-01 8.9941e-01 6.5820e-01]
15:46:32 E              [9.7412e-01 7.8979e-02 1.7737e-01]
15:46:32 E              [6.5625e-01 4.0503e-01 4.7058e-02]
15:46:32 E              [6.2939e-01 8.3789e-01 9.0332e-01]]
15:46:32 E           
15:46:32 E             [[3.7939e-01 3.0518e-02 1.1249e-01]
15:46:32 E              [3.2886e-01 3.9844e-01 2.6685e-01]
15:46:32 E              [6.5723e-01 3.1104e-01 2.1204e-01]
15:46:32 E              [5.0684e-01 8.4131e-01 4.4019e-01]
15:46:32 E              [6.7285e-01 2.7344e-01 4.6606e-01]
15:46:32 E              [3.2910e-01 1.0895e-01 4.0283e-01]]
15:46:32 E           
15:46:32 E             [[7.1924e-01 6.7383e-01 7.9834e-01]
15:46:32 E              [7.2705e-01 4.8438e-01 6.8652e-01]
15:46:32 E              [8.4351e-02 2.7515e-01 4.8218e-01]
15:46:32 E              [7.4463e-01 4.1895e-01 8.1909e-02]
15:46:32 E              [1.3098e-01 2.6807e-01 9.9072e-01]
15:46:32 E              [6.4551e-01 7.6611e-01 2.7075e-01]]
15:46:32 E           
15:46:32 E             [[7.2266e-02 4.7192e-01 9.6484e-01]
15:46:32 E              [4.7168e-01 3.7183e-01 5.1367e-01]
15:46:32 E              [6.7383e-01 7.3047e-01 7.8711e-01]
15:46:32 E              [8.2764e-01 7.6221e-01 2.8784e-01]
15:46:32 E              [5.5481e-02 7.0361e-01 3.7415e-02]
15:46:32 E              [8.1250e-01 7.0117e-01 8.1104e-01]]]]
15:46:32 E           directions[0]:
15:46:32 E           [[[[-1.32758054e-04  2.82382409e-02  1.08486095e-03  7.54827629e-02
15:46:32 E               -3.47154825e-02 -8.02623359e-02]
15:46:32 E              [-2.55459698e-02 -5.09881843e-02 -4.25160133e-02  8.53613597e-03
15:46:32 E               -7.91869503e-03 -1.21099188e-02]
15:46:32 E              [ 4.38838528e-02  2.21306036e-02 -6.08330834e-02  7.05122155e-02
15:46:32 E                2.30210075e-02  4.19067984e-02]
15:46:32 E              [-6.86766031e-04 -1.25234551e-02 -3.19515187e-02  6.79291011e-02
15:46:32 E               -1.21223248e-01  9.13102536e-04]
15:46:32 E              [-7.37402066e-02 -3.69493808e-02 -5.62313222e-04  2.45908837e-02
15:46:32 E                3.73570051e-02  3.70480057e-02]
15:46:32 E              [ 1.11889387e-01 -4.24881332e-02  1.83596466e-02  7.85051193e-02
15:46:32 E                3.32243199e-02 -1.30251267e-01]
15:46:32 E              [ 1.35235099e-02 -3.94264588e-02  6.05166294e-02 -4.34040607e-02
15:46:32 E                1.04782777e-01  4.29769421e-04]
15:46:32 E              [-2.83262717e-02  3.20803359e-02  7.88123781e-02 -4.16033265e-02
15:46:32 E                4.85544869e-02 -2.91591209e-03]]
15:46:32 E           
15:46:32 E             [[ 3.41771405e-02  7.91852652e-03 -7.29901141e-02 -3.46310619e-02
15:46:32 E               -2.55039268e-02 -1.09365820e-02]
15:46:32 E              [ 1.91027111e-03  1.15699951e-03  1.02487523e-02  6.97591509e-02
15:46:32 E               -9.42218316e-02 -5.33408188e-02]
15:46:32 E              [-7.51626617e-02 -2.85130128e-02  3.01082210e-02  2.07126448e-02
15:46:32 E               -6.96594036e-02  3.42265608e-02]
15:46:32 E              [-1.13803260e-02  5.04976383e-02  7.97469288e-02 -2.33241314e-02
15:46:32 E                6.43001097e-02  3.62031852e-02]
15:46:32 E              [ 9.22666131e-02 -2.28716190e-02  9.85181216e-02  1.19454934e-03
15:46:32 E               -5.10130506e-02 -9.33899481e-02]
15:46:32 E              [ 3.82373405e-02 -6.82228855e-02 -6.51511458e-02 -3.78682930e-02
15:46:32 E               -2.51001010e-02  7.67413787e-02]
15:46:32 E              [ 3.55297233e-02  1.27202390e-02  2.67084161e-02 -5.92149275e-02
15:46:32 E                7.76559165e-02  8.82211413e-02]
15:46:32 E              [-9.59639573e-02  4.81255862e-02  1.03100053e-01  8.08340887e-02
15:46:32 E               -4.04237303e-02  1.34274359e-02]]
15:46:32 E           
15:46:32 E             [[ 1.41527842e-02  2.56545524e-02 -2.61714409e-02  5.95207295e-02
15:46:32 E               -6.37904844e-02  4.01680578e-02]
15:46:32 E              [-8.17635543e-02 -8.34697541e-02  2.21280075e-03  6.87192154e-02
15:46:32 E               -5.43327871e-02 -3.54507010e-02]
15:46:32 E              [ 3.33649353e-02  4.02011436e-02 -5.33824121e-02 -8.34638783e-03
15:46:32 E               -4.00149269e-02 -4.99843270e-02]
15:46:32 E              [ 5.14074959e-02 -1.25067213e-02  6.02541551e-02  6.06757394e-02
15:46:32 E               -1.74885479e-02  3.27253440e-02]
15:46:32 E              [ 5.65431094e-02  9.68853944e-02 -5.10064319e-03 -1.45063743e-02
15:46:32 E               -4.34081811e-02  6.07554473e-03]
15:46:32 E              [-3.78689511e-03 -3.82870480e-02  9.44926251e-03  2.69743131e-03
15:46:32 E               -2.84673322e-02 -7.48796501e-02]
15:46:32 E              [ 2.64863230e-02  1.53421067e-01 -3.37259492e-02  7.33671893e-02
15:46:32 E               -7.53444046e-02 -8.36145208e-02]
15:46:32 E              [ 3.18095791e-02 -1.32756077e-03 -7.81350310e-02  1.30549592e-02
15:46:32 E                3.60549216e-02  5.69115863e-02]]]
15:46:32 E           
15:46:32 E           
15:46:32 E            [[[-5.72687130e-03 -6.35552405e-02 -9.96084272e-02 -3.05882005e-02
15:46:32 E                4.44132773e-02  9.85794145e-02]
15:46:32 E              [-8.77676549e-02 -3.67947550e-02  1.26380625e-02  5.91439160e-02
15:46:32 E               -1.11922045e-01  3.48821609e-02]
15:46:32 E              [-1.62367900e-02 -1.06715966e-01  3.02240577e-02  1.10907969e-03
15:46:32 E               -1.45447531e-01  1.13795200e-02]
15:46:32 E              [-6.14769227e-03  7.38893486e-02  5.88703930e-02  5.46129138e-03
15:46:32 E               -4.42491282e-02 -4.93650880e-02]
15:46:32 E              [ 5.64370625e-03  7.65299776e-03  1.29009750e-02  2.41580644e-02
15:46:32 E               -4.96855861e-02  3.39489469e-02]
15:46:32 E              [ 2.15806031e-02  9.68423307e-02 -1.06600628e-01 -1.78402028e-02
15:46:32 E                6.86918289e-02  1.01236755e-02]
15:46:32 E              [-3.23340564e-02  9.82923998e-02 -7.51269578e-02 -2.64276621e-02
15:46:32 E                6.73840722e-02 -3.00287755e-02]
15:46:32 E              [-4.63407424e-03 -1.22929789e-02  8.79780231e-02  4.06695495e-02
15:46:32 E                3.89297580e-02  1.06282246e-02]]
15:46:32 E           
15:46:32 E             [[ 2.23164032e-01  9.57106167e-02  5.82720495e-03  9.12576287e-03
15:46:32 E                2.42516162e-02 -3.03884327e-02]
15:46:32 E              [ 5.47136101e-02 -8.24500144e-02 -1.30386969e-02  2.16558747e-02
15:46:32 E               -1.12268492e-01 -4.84730541e-02]
15:46:32 E              [-7.87890103e-02 -7.94113995e-02  2.26778678e-02 -3.39933558e-02
15:46:32 E                3.46554139e-02  1.25905005e-02]
15:46:32 E              [-5.12887630e-02  1.60656647e-01 -9.70506708e-03 -9.70906513e-02
15:46:32 E               -3.76020445e-02 -1.43728921e-01]
15:46:32 E              [-5.72841602e-02  1.04329745e-02  1.37930967e-01 -7.43576468e-02
15:46:32 E                3.03668079e-02 -8.22868122e-02]
15:46:32 E              [-1.84122356e-02  3.39095855e-02 -6.12743535e-02  1.16030456e-01
15:46:32 E                7.36040974e-02  1.60989595e-02]
15:46:32 E              [-6.21547846e-02 -3.93048612e-02  2.65619401e-02  7.61019121e-03
15:46:32 E               -9.41173784e-02  2.79989540e-02]
15:46:32 E              [-9.28094101e-02 -7.92095934e-02  8.07864493e-02 -1.37045080e-01
15:46:32 E                1.25262295e-02  8.64667989e-02]]
15:46:32 E           
15:46:32 E             [[-9.70157179e-02  8.35286333e-02 -4.00196516e-02  6.41966293e-02
15:46:32 E                1.73372657e-02 -5.00959253e-02]
15:46:32 E              [ 1.84728889e-02 -1.12550093e-01 -2.10370280e-02  5.70346013e-02
15:46:32 E               -2.36156787e-02  2.23834476e-02]
15:46:32 E              [-3.44583197e-02  8.01812237e-02  1.14098425e-01 -5.83576024e-02
15:46:32 E               -4.17439855e-02  5.68681275e-04]
15:46:32 E              [ 3.09309831e-02  3.68441846e-02  4.41628394e-02 -4.77779789e-02
15:46:32 E               -6.43383964e-02 -7.88667019e-02]
15:46:32 E              [ 2.19364863e-03  7.93253407e-02 -1.90309694e-02 -5.96361795e-02
15:46:32 E               -1.62027639e-03 -3.69287576e-02]
15:46:32 E              [-1.98823115e-02  2.63491300e-02 -1.11272867e-02  7.37874108e-02
15:46:32 E               -2.47559301e-02  2.25956100e-02]
15:46:32 E              [-3.08116426e-02  9.14919787e-03  9.61579092e-02 -8.47518042e-03
15:46:32 E                8.53907585e-02  3.47457343e-02]
15:46:32 E              [ 2.01067435e-03 -3.04967542e-02  8.04134911e-02  1.55760730e-02
15:46:32 E                2.93159456e-02 -7.29616962e-03]]]]
15:46:32 E           gradients (numeric):  0.005823344490661706
15:46:32 E           gradients (backward): 0.0069573524451045365
15:46:32 E           
15:46:32 E           x: numeric gradient, y: backward gradient
15:46:32 E           Not equal to tolerance rtol=0.01, atol=0.001
15:46:32 E           
15:46:32 E           Mismatch: 100%
15:46:32 E           Max absolute difference: 0.00113401
15:46:32 E           Max relative difference: 0.16299418
15:46:32 E            x: array(0.005823)
15:46:32 E            y: array(0.006957)
15:46:32 E           
15:46:32 E           assert_allclose failed: 
15:46:32 E             shape: () ()
15:46:32 E             dtype: float64 float64
15:46:32 E             i: (0,)
15:46:32 E             x[i]: 0.005823344490661706
15:46:32 E             y[i]: 0.0069573524451045365
15:46:32 E             relative error[i]: 0.16299417966683039
15:46:32 E             absolute error[i]: 0.0011340079544428306
15:46:32 E             relative tolerance * |y[i]|: 6.957352445104537e-05
15:46:32 E             absolute tolerance: 0.001
15:46:32 E             total tolerance: 0.0010695735244510454
15:46:32 E           x: 0.00582334
15:46:32 E           y: 0.00695735
15:46:32 
15:46:32 atol       = 0.001
15:46:32 d          = array([[[[-1.32758054e-04,  2.82382409e-02,  1.08486095e-03,
15:46:32            7.54827629e-02, -3.47154825e-02, -8.02623359e-...   [ 2.01067435e-03, -3.04967542e-02,  8.04134911e-02,
15:46:32            1.55760730e-02,  2.93159456e-02, -7.29616962e-03]]]])
15:46:32 directions = [array([[[[-1.32758054e-04,  2.82382409e-02,  1.08486095e-03,
15:46:32            7.54827629e-02, -3.47154825e-02, -8.02623359e...  [ 2.01067435e-03, -3.04967542e-02,  8.04134911e-02,
15:46:32            1.55760730e-02,  2.93159456e-02, -7.29616962e-03]]]])]
15:46:32 eps        = 0.001
15:46:32 f          = <_io.StringIO object at 0x7f88929a3828>
15:46:32 gx_backward = array(0.00695735)
15:46:32 gx_numeric = array(0.00582334)
15:46:32 gy         = array([[[[7.2327e-02, 8.7646e-01, 9.4092e-01],
15:46:32          [8.9307e-01, 2.7905e-01, 7.8906e-01],
15:46:32          [6.0059e-01, 1....4e-01],
15:46:32          [5.5481e-02, 7.0361e-01, 3.7415e-02],
15:46:32          [8.1250e-01, 7.0117e-01, 8.1104e-01]]]], dtype=float16)
15:46:32 gys        = (array([[[[7.2327e-02, 8.7646e-01, 9.4092e-01],
15:46:32          [8.9307e-01, 2.7905e-01, 7.8906e-01],
15:46:32          [6.0059e-01, 1...-01],
15:46:32          [5.5481e-02, 7.0361e-01, 3.7415e-02],
15:46:32          [8.1250e-01, 7.0117e-01, 8.1104e-01]]]], dtype=float16),)
15:46:32 i          = 0
15:46:32 rtol       = 0.01
15:46:32 self       = <chainer.gradient_check._CheckBackward object at 0x7f88929616a0>
15:46:32 x          = array([[[[0.4407  , 0.4185  , 0.778   , 0.8325  , 0.4678  , 0.47    ],
15:46:32          [0.884   , 0.441   , 0.4668  , 0.4902 ...  , 0.7715  , 0.569   ],
15:46:32          [0.00766 , 0.6416  , 0.973   , 0.536   , 0.0473  , 0.9062  ]]]],
15:46:32       dtype=float16)
15:46:32 xs         = (array([[[[0.4407  , 0.4185  , 0.778   , 0.8325  , 0.4678  , 0.47    ],
15:46:32          [0.884   , 0.441   , 0.4668  , 0.4902..., 0.7715  , 0.569   ],
15:46:32          [0.00766 , 0.6416  , 0.973   , 0.536   , 0.0473  , 0.9062  ]]]],
15:46:32       dtype=float16),)
15:46:32 
15:46:32 chainer/gradient_check.py:537: AssertionError
```

--- a/tests/chainer_tests/functions_tests/array_tests/test_im2col.py
+++ b/tests/chainer_tests/functions_tests/array_tests/test_im2col.py
@@ -118,7 +118,7 @@ class TestIm2Col(unittest.TestCase):
 
         self.check_backward_options = {'atol': 5e-4, 'rtol': 5e-3}
         if self.dtype is numpy.float16:
-            self.check_backward_options.update({'atol': 1e-3, 'rtol': 1e-2})
+            self.check_backward_options.update({'atol': 2e-3, 'rtol': 1e-2})
 
         self.check_double_backward_options = {'atol': 5e-4, 'rtol': 5e-3}
         if self.dtype is numpy.float16:

